streams/src/main/java/org/apache/kafka/streams/kstream/Consumed.java

```diff
@@ -96,13 +112,21 @@ protected Consumed(final Consumed<K, V> consumed) {
      *
      * @return a new instance of {@link Consumed}
      */
+    @Deprecated
     public static <K, V> Consumed<K, V> with(final Serde<K> keySerde,
                                              final Serde<V> valueSerde,
                                              final TimestampExtractor timestampExtractor,
                                              final Topology.AutoOffsetReset resetPolicy) {
         return new Consumed<>(keySerde, valueSerde, timestampExtractor, resetPolicy, null);
```

Comment:
We need to update this call to pass in one more paremeter:
```
return new Consumed<>(keySerde, valueSerde, timestampExtractor, resetPolicy, convertOldToNew(resetPolicy), null);
```

plus add a helper:
```
private static AutoOffsetReset convertOldToNew(final TopologyAutoOffsetReset resetPolicy) {
    if (resetPolicy == null) {
        return null;
    }

    return resetPolicy = EARLIEST ? AutoOffsetReset.earliest() : AutoOffsetReset.latest();
}
```

Reply:
Updated, please see below comment as I am not 100% on how I've implemented this.